File: PRACTICA3/Cola.java
public class Cola {
    private long[] arreglo;
    private int inicio;
    private int fin;
    private int n;
    private int size;

    public Cola(int n) {
        this.n = n;
        arreglo = new long[n]; 
        inicio = 0;            
        fin = 0;               
        size = 0;              
    }

    public void insert(long e) throws Exception {
        if (!isFull()) {
            arreglo[fin] = e;
            fin = (fin + 1) % n;
            size++;
            System.out.println("Elemento " + e + " añadido a la cola.");
        } else {
            throw new Exception("Cola llena");
        }
    }

    public long remove() throws Exception {
        if (!isEmpty()) {
            long elemento = arreglo[inicio];
            inicio = (inicio + 1) % n;
            size--;
            System.out.println("Elemento " + elemento + " eliminado de la cola.");
            return elemento;
        } else {
            throw new Exception("Cola vacía");
        }
    }

    public long peek() throws Exception {
        if (!isEmpty()) {
            return arreglo[inicio];
        } else {
            throw new Exception("Cola vacía");
        }
    }

    public boolean isEmpty() {
        return size == 0;
    }

    public boolean isFull() {
        return size == n;
    }

    public int size() {
        return size;
    }

    public static void main(String[] args) {
        try {
            Cola cola = new Cola(5);
            cola.insert(9);
            cola.insert(11);
            System.out.println("-----------------"); 
            System.out.println("Elemento en la cabeza: " + cola.peek()); 
            System.out.println("Elemento retirado: " + cola.remove());
            System.out.println("-----------------");   
            System.out.println("¿Está la cola vacía? " + cola.isEmpty()); 
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }
    }
}

File: PRACTICA3/Cola.py
class Cola:
    def _init_(self, n):
        self.arreglo = [0] * n  
        self.inicio = 0          
        self.fin = 0             
        self.n = n               
        self.size = 0            

    def insert(self, e):
        if not self.isFull():
            self.arreglo[self.fin] = e
            self.fin = (self.fin + 1) % self.n
            self.size += 1
            print(f'Elemento {e} añadido a la cola.')
        else:
            raise Exception("Cola llena")

    def remove(self):
        if not self.isEmpty():
            elemento = self.arreglo[self.inicio]
            self.inicio = (self.inicio + 1) % self.n
            self.size -= 1
            print(f'Elemento {elemento} eliminado de la cola.')
            return elemento
        else:
            raise Exception("Cola vacía")

    def peek(self):
        if not self.isEmpty():
            return self.arreglo[self.inicio]
        else:
            raise Exception("Cola vacía")

    def isEmpty(self):
        return self.size == 0

    def isFull(self):
        return self.size == self.n

    def size(self):
        return self.size

if __name__ == "_main_":
    cola = Cola(3)  
    cola.insert(5)
    cola.insert(6)
    print(f'Elemento en la cabeza: {cola.peek()}')  
    print(f'Elemento retirado: {cola.remove()}')    
    print(f'¿Está la cola vacía? {cola.isEmpty()}')  